src/lib/core/immer-adapter/common/immutable-state-context.ts

```diff
@@ -39,4 +46,42 @@ export class ImmutableStateContext<T extends object> implements StateContext<T>
   public dispatch(actions: any | any[]): Observable<void> {
     return this.ctx.dispatch(actions);
   }
+
+  private clone<T = any>(obj: T): T {
+    let copy: any;
+
+    // Handle the 3 simple types, and null or undefined
+    if (null == obj || 'object' !== typeof obj) {
```

Comment:
I guess `obj == null || typeof obj !== 'object'` looks better :D

Reply:
Yes, I just copied the code without looking)